pubsub/src/lib.rs

```diff
@@ -113,13 +211,24 @@
 //!     // Or simply use the `subscription.subscribe` method.
 //!     subscription.receive(|mut message, cancel| async move {
 //!         // Handle data.
-//!         let data = message.message.data.as_ref();
-//!         println!("{:?}", data);
+//!         println!("Got Message: {:?}", message.message.data);
 //!
 //!         // Ack or Nack message.
 //!         message.ack().await;
 //!     }, cancel.clone(), None).await;
 //!
+//!     // Alternativly you can use the messages as a stream
+//!     // (needs futures_util::StreamExt as import)
+//!     // Note: This blocks the current thread but helps working with non clonable data
+//!     let mut stream = subscription.subscribe(None).await?;
+//!     while let Some(message) = stream.next().await {
```

Comment:
Since doc test does not terminate forever, the sample code using MessageStream should be a separate function from the existing one.

Reply:
You are right.
I moved the example to a seperate code block and marked it as `no_run` so we dont get stuck in the tests.